File: Container With Most Water.cpp
class Solution {
public:
    /**
     * @param heights: a vector of integers
     * @return: an integer
     */
    int maxArea(vector<int> &heights) {
        // write your code here
        if (heights.empty())
        {
            return 0;
        }
        int i = 0, j = heights.size() - 1;
        int max_area = INT_MIN;
        while (i < j)
        {
            max_area = max(max_area, min(heights[i], heights[j]) * (j - i));
            if (heights[i] > heights[j])
            {
                --j;
            }
            else
            {
                ++i;
            }
        }
        return max_area;
    }
};

File: Container With Most Water.py
class Solution:
    # @param heights: a list of integers
    # @return: an integer
    def maxArea(self, heights):
        # write your code here
        if not heights:
            return 0
        
        length = len(heights)
        left, right = 0, length - 1
        ret = 0
        while left < right:
            if heights[left] > heights[right]:
                ret = max(ret, heights[right] * (right - left))
                right -= 1
            else:
                ret = max(ret, heights[left] * (right - left))
                left += 1
        return ret

        